### PR TITLE
Only require backports.functools_lru_cache for python < 3.3

### PR DESCRIPTION
**Pull request by dcermak**
❓ **What kind of change does this PR introduce?**
  - [x] 🐞 bug fix
  - [ ] 🐣 feature
  - [ ] 📋 docs update
  - [ ] 📋 tests/coverage improvement
  - [ ] 📋 refactoring
  - [ ] 💥 other

📋 **What is the related issue number (starting with `#`)**

None

❓ **What is the current behavior?** (You can also link to an open issue here)

cheroot requires the `backports.functools_lru_cache` module even for python versions that ship `functools.lru_cache`

❓ **What is the new behavior (if this is a feature change)?**

cheroot only requires `backports.functools_lru_cache` for python < 3.3

📋 **Other information**:

📋 **Checklist**:

  - [x] I think the code is well written
  - [x] I wrote [good commit messages][1]
  - [ ] I have [squashed related commits together][2] after the changes have been approved
  - [-] Unit tests for the changes exist
  - [-] Integration tests for the changes exist (if applicable)
  - [-] I used the same coding conventions as the rest of the project
  - [-] The new code doesn't generate linter offenses
  - [-] Documentation reflects the changes
  - [-] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences

[1]: http://chris.beams.io/posts/git-commit/
[2]: https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

---
This change is [Reviewable](https://reviewable.io/reviews/cherrypy/cheroot/221)

### CODE REVIEW

**Comment by smithfarm** [comment]
@webknjaz Was the "wontfix" label added intentionally?

**Comment by webknjaz** [comment]
@smithfarm it's not broken. You didn't install the dependency.

**Comment by webknjaz** [comment]
I'm leaving this up to @jaraco to decide whether to merge this. It's a duplicate of what he blocked two times already.

**Comment by smithfarm** [comment]
> @smithfarm it's not broken. You didn't install the dependency.

But the dependency is not needed for cheroot to work on recent Python 3, so why have it?

**Comment by webknjaz** [comment]
@smithfarm it's a fallback in runtime. But `pkg_resources` seems to disagree. That dep is noop under Python 3.

**Comment by webknjaz** [comment]
Anyway, I'm almost sure that you didn't use Pip to install Cheroot which means that you must blame your OS packager.

**Comment by webknjaz** [comment]
It's a duplicate of https://github.com/cherrypy/cheroot/issues/181 and https://github.com/cherrypy/cheroot/issues/203

**Comment by smithfarm** [comment]
> Anyway, I'm almost sure that you didn't use Pip to install Cheroot which means that you must blame your OS packager.

Right, I am trying to keep this stuff running in the OS.

If this fix helps the OS maintainers and does not hurt folks using pip, I don't understand why it's being blocked?

**Comment by webknjaz** [comment]
https://github.com/cherrypy/cheroot/commit/ea8acd0786cfb7435011f018897fbfd0f9759415

**Comment by webknjaz** [comment]
Feel free to read all the discussions in linked tickets. But the bottom line is that we only support Pip and separate ourselves from anything that is related to OS cases because it's something that somebody else who is interested in this should maintain.

**Comment by dcermak** [comment]
> But the bottom line is that we only support Pip and separate ourselves from anything that is related to OS cases because it's something that somebody else who is interested in this should maintain.

I fully understand that.

What I don't understand is why you insist on keeping `backports.functools_lru_cache` as a direct dependency for Python versions that ship this module in their standard library? Yes, the installation is not broken with pip (because it just downloads the package but never uses it) and yes we can work around this dependency when creating a rpm package, but that's not the point of this pull request. The point is that you **do not need** `backports.functools_lru_cache` as `lru_cache` is part of `functools` since Python 3.2. So why require it then?

**Comment by jaraco** [comment]
That reversion commit mentions #85.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,3 +1,9 @@
+v6.6.0
+======
+
+- Revisit :pr:`85` under :pr:`221`. Now ``backports.unittest_mock``
```
Comment (dcermak) [inline CHANGES.rst]:
```suggestion
- Revisit :pr:`85` under :pr:`221`. Now ``backports.functools_lru_cache``
```

Comment (jaraco) [reply CHANGES.rst]: Added in 2f6199ae

**Comment by jaraco** [comment]
> So why require it then?

Reading up on #85, which goes into detail, the tl;dr is that it's simpler for this project (and all other projects) to unconditionally declare these compatibility libraries and in fact, the backports guidance specifically advises projects to [unconditionally _use_ the backport](https://pypi.org/project/backports/) (see "What if the feature is present"). Cheroot doesn't do that, which is what gives the impression that it's an optional dependency. It's the declared dependencies that are authoritative, so if downstream packagers aren't honoring the declared dependencies, that's a bug in their system.

So let's say we accept this proposal. Does that make it easier for downstream packagers? It arguably makes it more difficult, as now packagers have different dependency trees depending on the target Python version.

Still, since cheroot is conditionally importing this dependency, I guess it's fine to accept to make it easier on the packagers.

**Comment by dcermak** [comment]
> Does that make it easier for downstream packagers?

Thank you for merging this, it actually makes our job easier in opensuse.

**Pull request merged**